sky/packages/sky/lib/src/widgets/navigator.dart

```diff
@@ -330,6 +326,71 @@ class NavigatorState extends State<Navigator> {
 
 }
 
+class _RouteWidget extends StatefulComponent {
+  _RouteWidget({
+    Route route,
+    this.nextRoute,
+    this.buildTargetHeroes: false
+  }) : route = route,
+       super(key: new ObjectKey(route));
+  final Route route;
+  final Route nextRoute;
+  final bool buildTargetHeroes;
+  _RouteWidgetState createState() => new _RouteWidgetState();
+}
+
+class _RouteWidgetState extends State<_RouteWidget> {
+  void initState() {
+    super.initState();
+    config.route._widgetState = this;
```

Comment:
(I think this stuff can certainly be further rationalised, but that's something we should do in a separate patch.)